slack-pi-command/README.md

```diff
@@ -0,0 +1,171 @@
+# Slack π command (NodeJS)
+
+A slack command that returns π to _n_ digits, via a Binaris function.
+
+
+### Table of Contents
+1. [Return _n_ digits of π with a Binaris Function](#calculate-pi)
+1. [Integrate with Slack](#integrate-slack)
+
+<a name="calculate-pi"></a>
+
+## Return _n_ digits of π with a Binaris Function
+
+It is assumed you already have a Binaris account. If you don't have an account yet, worry not, signing up is painless and takes just 2 minutes. Visit [Getting Started](https://dev.binaris.com/tutorials/nodejs/getting-started/) to find concise instructions for the process.
+
+> Note: To complete this tutorial you will need administrator permissions on your Slack workspace.
+
+We'll start by creating the skeleton for our π function.
+
+```bash
+bn create node8 public_slack_pi_command
+```
+
+It's important the function name begin with `public_`, this indicates to the Binaris backend that the function should be publicly accessible.
+
+The first change we'll make, is to parse and convert the input payload Slack sends into something usable. Slack sends command payloads as `x-url-form-encoded` which means we'll need to convert it into a native JavaScript object. Luckily the `query-string` package from `npm` can do a lot of our heavy lifting for us.
+
+> Note: Binaris will attempt to `JSON.parse` any incoming payloads, successfully parsed payloads reside in the `body` field.
+
+```diff
++const { parse } = require('query-string');
+
+ exports.handler = async (body, context) => {
++  const parsed = parse(context.body.toString('utf8'));
+-  const name = context.request.query.name || body.name || 'World';
+-  return `Hello ${name}!`;
+ };
+```
+
+Now that we have the payload in a usable form, let's extract the data we need, so we compute π to the number of digits requested from our user.
+
+```diff
+   const parsed = parse(context.body.toString('utf8'));
++  const numDigits = parseInt(parsed.text, 10);
+ };
+```
+
+Now let's add some generic validation
+
+```diff
+   const numDigits = parseInt(parsed.text, 10);
++
++  // ensure the input is a valid, natural number
++  if (isNaN(numDigits) || numDigits <= 0) {
++    return {
++      text: `Expected non-zero, natural number, received "${parsed.text}"`,
++      response_type: 'ephemeral',
++    };
++  }
+
++  // enforce a fun limit on the max pi digits
++  if (numDigits > 10) {
++    return {
++      text: "You've already had too much 🥧!",
++      response_type: 'ephemeral',
++    };
++  }
+ };
+```
+
+> Note: We return the specific fields `text` & `response_type` because this is the expected format of an "Error" response in the Slack API.
+
+All that's left to do is return π to the number of digits requested by the user. The `npm` package `pi` makes this very convenient.
+
+
+```diff
+   }
++  return PI(numDigits - 1);
+};
+
+```
+
+> Note: We subtract one from the input value because the first 3 in π is not considered a "digit"
```

Comment:
is this material? consider simply using numDigits and dropping this altogether. will probably make the reader's life easier.

Reply:
Avner specifically asked me to add it.

Reply:
Avner specifically asked me to add it.